### PR TITLE
Revamp build pipeline

### PR DESCRIPTION
**Pull request by sndrs**
- Removes support for custom `template.html`, but…
- Adds support for svelte’s [head component](https://svelte.dev/tutorial/svelte-head)
- Nested islands are deduplicated in client bundles
- Removes the `Island` component (islands are now imported like any other svelte component)
- Build pipeline is managed by ESBuild plugins

### CODE REVIEW

File: src/esbuild_plugins/build_routes/_template.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<meta charset="UTF-8" />
+		<meta http-equiv="X-UA-Compatible" content="IE=edge" />
+		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+		<title>Svelte + Deno</title>
+		<!-- MNNYKS:JS -->
+		<!-- MNNYKS:CSS -->
```
Comment (mxdvl) [inline src/esbuild_plugins/build_routes/_template.html]: Is `Y` a vowel?

**Review by mxdvl — APPROVED** [review]
Great!

File: src/esbuild_plugins/build_routes/_template.html (hunk 1 above)
Comment (mxdvl) [reply src/esbuild_plugins/build_routes/_template.html]: This whole file will be gone in #19

**Comment by mxdvl** [comment]
Good to see that the deploy still worked fine! https://mononykus-86g9126sk4s0.deno.dev/

File: src/esbuild_plugins/build_routes/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	setup(build) {
+		build.onEnd(async (result) => {
+			const start = performance.now();
+
+			const routes = result.outputFiles ?? [];
+
+			for (const route of routes) {
+				const module = await import(
+					"data:application/javascript," + encodeURIComponent(route.text)
+				) as {
```
Comment (mxdvl) [inline src/esbuild_plugins/build_routes/index.ts:25]: This is still amazing to me!

https://github.com/denoland/deno/issues/5059

Comment (sndrs) [reply src/esbuild_plugins/build_routes/index.ts:25]: yeah this is a much nicer way of doing it

**Review by mxdvl — APPROVED** [review]

**Pull request merged**